core/src/types/buffer.rs

```diff
@@ -317,6 +317,32 @@ impl Buffer {
             }
         }
     }
+
+    /// Splits the current object into two parts at the specified index (`at`).
+    ///
+    /// The portion before the index is stored in the `left` object, while the portion
+    /// after the index is removed from the current object (`self`).
+    #[inline]
+    pub fn split_to(&mut self, at: usize) -> Self {
```

Comment:
Hi, would you like to add test cases for them?

Reply:
Kindly cc @hanxuanliang for a look.

Reply:
I'm sorry that the issue has been delayed recently due to personal matters. I will finish the following pr content at the weekend

Reply:
> I'm sorry that the issue has been delayed recently due to personal matters. I will finish the following pr content at the weekend

There's no hurry; I just want an update on the current status. You can return at any time.

Reply:
I have been sailing in the A-stock market for a while, and finally have some time to pay off the accumulated debt of Rust. Wait for me!